datasets/dataset_imdb.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+"""
+IMDB dataset.
+inspired from:
+https://keras.io/examples/nlp/text_classification_from_scratch/
+https://builtin.com/data-science/how-build-neural-network-keras
+"""
+
+import tensorflow as tf
+import numpy as np
+from tensorflow import keras
+from keras.utils import to_categorical
+from keras import models
+from keras import layers
+from keras.datasets import imdb
+
+(x_train, y_train), (x_test, y_test) = imdb.load_data(num_words=10000, skip_top=35, start_char=1, oov_char=2, index_from=3)
+
+# Argument num_words => only the num_words first more frequent words are used
+# The others are replaced by 2
+# A Review start by the number 1
+# The longest review is less than 3000 words long
+
+def preprocess_dataset_inputs(data):
+    data_list = []
+    for sample in data:
+        sample += [2] * (nb_words - len(sample)) # 2 is the over_of_vocabulary number
+        data_list.append(sample)
+
+    data_array = np.array(data_list)
+    data_array = data_array.reshape(data_array.shape[0], data_array.shape[1], 1)
+
+    return data_array
+
+def preprocess_dataset_labels(y): # Do not preprocess labels here! => it's done in another script
+    y = keras.utils.to_categorical(y, num_classes)
+
+    return y
+
+# Padding
+nb_words = 3000
+
+input_shape = (nb_words,1)
```

Comment:
Hmmm, maybe add a check in the padding function that verifies that the longest review is shorter than the `nb_words` parameter you impose.

Reply:
There is now an `assert` in the function `preprocess_dataset_inputs`